bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/DbLedgerStorageConfiguration.java

```diff
@@ -0,0 +1,161 @@
+
+    private static final long DEFAULT_DIRECT_IO_TOTAL_READBUFFER_SIZE_MB =
+            (long) (0.125 * PlatformDependent.estimateMaxDirectMemory()) / MB;
+
+    private static final String DIRECT_IO_ENTRYLOGGER_READBUFFER_SIZE_MB =
+            "dbStorage_directIOEntryLoggerReadBufferSizeMb";
+
+    private static final long DEFAULT_DIRECT_IO_READBUFFER_SIZE_MB = 8;
+
+    private static final String DIRECT_IO_ENTRYLOGGER_MAX_FD_CACHE_TIME_SECONDS =
+            "dbStorage_directIOEntryLoggerMaxFdCacheTimeSeconds";
+
+    private static final int DEFAULT_DIRECT_IO_MAX_FD_CACHE_TIME_SECONDS = 300;
+
+    @VisibleForTesting
+    public static final String MAX_THROTTLE_TIME_MILLIS = "dbStorage_maxThrottleTimeMs";
+
+    private static final long DEFAULT_MAX_THROTTLE_TIME_MILLIS = TimeUnit.SECONDS.toMillis(10);
+
+    public long getWriteCacheMaxSize() {
+        return getLongVariableOrDefault(WRITE_CACHE_MAX_SIZE_MB, DEFAULT_WRITE_CACHE_MAX_SIZE_MB) * MB;
+    }
+
+    public long getReadCacheMaxSize() {
+        return getLongVariableOrDefault(READ_AHEAD_CACHE_MAX_SIZE_MB, DEFAULT_READ_CACHE_MAX_SIZE_MB) * MB;
+    }
+
+    public int getReadAheadCacheBatchSize() {
+        return this.getInt(READ_AHEAD_CACHE_BATCH_SIZE, DEFAULT_READ_AHEAD_CACHE_BATCH_SIZE);
+    }
+
+    public boolean isDirectIOEntryLoggerEnabled() {
+        return getBooleanVariableOrDefault(DIRECT_IO_ENTRYLOGGER, false);
+    }
+
+    public long getDirectIOEntryLoggerTotalWriteBufferSize() {
+        return getLongVariableOrDefault(DIRECT_IO_ENTRYLOGGER_TOTAL_WRITEBUFFER_SIZE_MB,
+                DEFAULT_DIRECT_IO_TOTAL_WRITEBUFFER_SIZE_MB) * MB;
+    }
+
+    public long getDirectIOEntryLoggerTotalReadBufferSize() {
+        return getLongVariableOrDefault(DIRECT_IO_ENTRYLOGGER_TOTAL_READBUFFER_SIZE_MB,
+                DEFAULT_DIRECT_IO_TOTAL_READBUFFER_SIZE_MB) * MB;
+    }
+
+    public long getDirectIOEntryLoggerReadBufferSize() {
+        return getLongVariableOrDefault(DIRECT_IO_ENTRYLOGGER_READBUFFER_SIZE_MB, DEFAULT_DIRECT_IO_READBUFFER_SIZE_MB)
+                * MB;
+    }
+
+    public long getDirectIOEntryLoggerMaxFDCacheTimeSeconds() {
+        return getLongVariableOrDefault(DIRECT_IO_ENTRYLOGGER_MAX_FD_CACHE_TIME_SECONDS,
+                DEFAULT_DIRECT_IO_MAX_FD_CACHE_TIME_SECONDS);
+    }
+
+    public long getMaxThrottleTimeMillis() {
+        return this.getLong(MAX_THROTTLE_TIME_MILLIS, DEFAULT_MAX_THROTTLE_TIME_MILLIS);
+    }
+
+    /**
+     * The configured pooling concurrency for the allocator, if user config it, we should consider the unpooled direct
+     * memory which readCache and writeCache occupy when use DbLedgerStorage.
+     */
+    public int getAllocatorPoolingConcurrency() {
+        long writeCacheSize = this.getWriteCacheMaxSize();
+        long readCacheSize = this.getReadCacheMaxSize();
+        long availableDirectMemory = PlatformDependent.maxDirectMemory() - writeCacheSize - readCacheSize;
```

Comment:
Please use `PlatformDependent.estimateMaxDirectMemory()` instead of `PlatformDependent.maxDirectMemory()`. Please refer to: https://github.com/apache/bookkeeper/pull/2989

Reply:
In dbLedgerStorage, 
- If we use `DefaultEntryLogger`, only the whole write-cache will be allocated at one time in the init state, and the read-cache will be allocated on-demand.
- If we use `DirectEnryLogger`, both the write-cache and the read-cache are allocated on-demand.

So I think we'd better keep the default implementation.